avocado/utils/podman.py

```diff
@@ -0,0 +1,104 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2021 Red Hat Inc.
+# Authors : Beraldo Leal <bleal@redhat.com>
+
+"""
+This module provides an basic API for interacting with podman.
+
+This module it was designed to be executed in async mode. Remember this when
+consuming this API.
+"""
+
+import json
+import logging
+import os
+
+from asyncio import create_subprocess_exec, subprocess
+
+
+LOG = logging.getLogger(__name__)
+
+
+class PodmanException(Exception):
+    pass
+
+
+class Podman:
+    def __init__(self, podman_bin):
+        if not os.path.exists(podman_bin):
```

Comment:
This is OK, and I understand it came from the existing spawner code, but as a suggestion: it can be further strengthened by checking for executable permissions (`os.access(podman_bin, os.R_OK | os.X_OK)`).

Reply:
BTW, we have `avocado.utils.path.check_readable()`, I think we could use a `is_readable_executable()` too and have it reused everywhere.

Reply:
maybe just use `shutil.which('python')` ?

Reply:
> maybe just use `shutil.which('python')` ?

So, I'm assuming that is the suggestion in place of `is_readable_executable()`.  For the `podman_bin` example, are you planning on making `podman_bin` optional and defaulting to `shutil.which('podman')`? I'd second that.

Reply:
And yes, I confused with the python method. :)

Reply:
Something like: 

```python
        path = which(podman_bin or 'podman') 
        if not path:
            msg = f"Podman binary {podman_bin} is not available on the system."
            raise PodmanException(msg)

```